src/sources/eventstoredb/types.rs

```diff
@@ -0,0 +1,276 @@
+#[serde(rename_all = "camelCase")]
+pub struct Proc {
+    pub id: usize,
+    pub mem: usize,
+    pub cpu: f64,
+    pub threads_count: i64,
+    pub thrown_exceptions_rate: f64,
+    pub disk_io: DiskIo,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct DiskIo {
+    pub read_bytes: usize,
+    pub written_bytes: usize,
+    pub read_ops: usize,
+    pub write_ops: usize,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Sys {
+    pub free_mem: usize,
+    pub loadavg: LoadAvg,
+    pub drive: Option<Drive>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct LoadAvg {
+    #[serde(rename = "1m")]
+    pub one_m: f64,
+    #[serde(rename = "5m")]
+    pub five_m: f64,
+    #[serde(rename = "15m")]
+    pub fifteen_m: f64,
+}
+
+#[derive(Debug)]
+pub struct Drive {
+    pub path: String,
+    pub stats: DriveStats,
+}
+
+impl<'de> Deserialize<'de> for Drive {
+    fn deserialize<D>(deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_map(DriveVisitor)
+    }
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct DriveStats {
+    pub available_bytes: usize,
+    pub total_bytes: usize,
+    pub usage: String,
+    pub used_bytes: usize,
+}
+
+struct DriveVisitor;
+
+impl<'de> Visitor<'de> for DriveVisitor {
+    type Value = Drive;
+
+    fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(formatter, "DriveStats object")
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, <A as MapAccess<'de>>::Error>
+    where
+        A: MapAccess<'de>,
+    {
+        if let Some(key) = map.next_key()? {
+            return Ok(Drive {
+                path: key,
+                stats: map.next_value()?,
+            });
+        }
+
+        Err(serde::de::Error::missing_field("<Drive path>"))
+    }
+}
+
+#[derive(Debug)]
+pub struct EventStoreDbMetricsHttpError {
+    pub error: crate::Error,
+}
+
+impl InternalEvent for EventStoreDbMetricsHttpError {
+    fn emit_logs(&self) {
+        error!(message = "HTTP request processing error.", error = ?self.error);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("http_request_errors_total", 1);
+    }
+}
+
+#[derive(Debug)]
+pub struct EventStoreDbStatsParsingError {
+    pub error: serde_json::Error,
+}
+
+impl InternalEvent for EventStoreDbStatsParsingError {
+    fn emit_logs(&self) {
+        error!(message = "JSON parsing error.", error = ?self.error);
+    }
+
+    fn emit_metrics(&self) {
+        counter!("stats_parsing_errors_total", 1);
+    }
+}
+
+pub struct EventStoreDbMetricsReceived;
+
+impl InternalEvent for EventStoreDbMetricsReceived {
+    fn emit_logs(&self) {
+        debug!("Stats scraped.");
+    }
+
+    fn emit_metrics(&self) {
+        counter!("processed_stats_total", 1);
```

Comment:
Instead of this metric `events_in_total` and `processed_bytes_total` should be emitted. Look at `PrometheusEventReceived` for example.

Reply:
done